xtra-libp2p-rollover/src/next/taker.rs

```diff
@@ -0,0 +1,368 @@
+                                        "Expected Msg2 within {} seconds",
+                                        ROLLOVER_MSG_TIMEOUT.as_secs()
+                                    )
+                                })?
+                                .context("Empty stream instead of Msg2")?
+                                .context("Unable to decode listener Msg2")?
+                                .into_rollover_msg()?
+                                .try_into_msg2()?;
+
+                            let revocation_sk_theirs = msg2.revocation_sk;
+                            let revoked_commits = dlc
+                                .base_dlc_params_from_latest(complete_fee_before_rollover)
+                                .revoke_base_commit_tx(revocation_sk_theirs)
+                                .context("Maker sent invalid revocation sk")?;
+
+                            let dlc = Dlc {
+                                identity: dlc.identity,
+                                identity_counterparty: dlc.identity_counterparty,
+                                revocation: rev_sk,
+                                revocation_pk_counterparty: punish_params.maker.revocation_pk,
+                                publish: publish_sk,
+                                publish_pk_counterparty: punish_params.maker.publish_pk,
+                                maker_address: dlc.maker_address,
+                                taker_address: dlc.taker_address,
+                                lock: dlc.lock.clone(),
+                                commit: (own_cfd_txs.commit.0.clone(), msg1.commit, commit_desc),
+                                cets,
+                                refund: (refund_tx, msg1.refund),
+                                maker_lock_amount: dlc.maker_lock_amount,
+                                taker_lock_amount: dlc.taker_lock_amount,
+                                revoked_commit: revoked_commits,
+                                settlement_event_id,
+                                refund_timelock: rollover_params.refund_timelock,
+                            };
+
+                            emit_completed(
+                                order_id,
+                                dlc,
+                                funding_fee,
+                                complete_fee.into(),
+                                &executor,
+                            )
+                            .await;
+                        }
+                        Decision::Reject(_) => {
+                            emit_rejected(order_id, &executor).await;
+                        }
+                    }
+                    Ok(())
+                }
+            },
+            {
+                let executor = self.executor.clone();
+                move |e| async move {
+                    emit_failed(order_id, e, &executor).await;
+                }
+            },
+        );
+    }
+}
```

Comment:
I did not check the logic in here. I guess this is all copy&paste?

Reply:
It should be a straight copy of the original `current`, apart from using `MaxInversePrice::OliviaMax`.